### PR TITLE
Proposal: Arm eslint with additional rules for async/await

### PR DESCRIPTION
**Pull request by LinqLover**
This might help us to avoid further bugs like #11.

See also the discussion in https://github.com/hpi-swa/setup-smalltalkCI/commit/a89a05efc1acb01977b6acf27bd1fa506cacaacb.

### CODE REVIEW

**Comment by LinqLover** [comment]
image

Sigh ...

File: index.js [hunk 1]
```diff
@@ -118,4 +118,5 @@ function isPlatform(image, name) {
   return image.toLowerCase().startsWith(name)
 }
 
-run()
+// eslint-disable-next-line no-floating-promise/no-floating-promise
+run()  // return a Promise as specified by the GitHub Actions protocol
```
Comment (fniephaus) [inline index.js:122]: should this line also have an await?

Comment (LinqLover) [reply index.js:122]: I looked up a few popular GitHub actions that are implemented in JavaScript and none of them had an `await` there. Perhaps the most popular example is this one:

https://github.com/actions/javascript-action/blob/2498a9f3f2a667476bb92072bc0e3fd9634572bc/index.js#L5

I have no idea why this is, and how the GitHub Actions engine is able to await these promises anyway, but apparently it's common practice. Also, I have only been using promises for ~2 weeks as of today. 😅

**Review by fniephaus — APPROVED** [review]
LGTM, thanks!

**Pull request merged**